src/lib/x509/certstor_system_windows/certstor_windows.cpp

```diff
@@ -0,0 +1,236 @@
+/*
+* Certificate Store
+* (C) 1999-2019 Jack Lloyd
+* (C) 2018-2019 Patrik Fiedler, Tim Oesterreich
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/certstor_windows.h>
+#include <botan/der_enc.h>
+
+#include <array>
+#include <vector>
+#include <iostream>
+
+#define NOMINMAX 1
+#define _WINSOCKAPI_ // stop windows.h including winsock.h
+#include <Windows.h>
+#include <Wincrypt.h>
+
+namespace {
+
+const std::array<std::string, 4> cert_store_names{"MY", "Root", "Trust", "CA"};
+
+HCERTSTORE openCertStore(const std::string& cert_store_name)
+   {
+   auto store = CertOpenSystemStore(0, cert_store_name.c_str());
```

Comment:
Use nullptr instead of 0/NULL

Reply:
`nullptr` doesn't seem to work...
The type of this parameter is `HCRYPTPROV_LEGACY` which is a typedef to `ULONG_PTR` which is a typedef to `unsigned __int3264` which is a (sort of?) integer type discussed [here](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/21eec394-630d-49ed-8b4a-ab74a1614611).

Reply:
That's terrifying. OK.